src/analysis/retail/deathknight/blood/components/ResourceWasteProblemRenderer.tsx

```diff
@@ -45,6 +45,8 @@ export function ResourceWasteProblemRenderer({
               return <>This cast wasted {this._wastedRp} RP.</>;
             },
           };
+          // TODO: Come up with a better way of handling this so that it doesn't break any existing meta
+          // eslint-disable-next-line wowanalyzer/event-meta-inefficient-cast
```

Comment:
so this is operating on a copy of events and intentionally overwriting the meta of the copy (see [this branch](https://github.com/WoWAnalyzer/WoWAnalyzer/pull/6783/files#diff-710f818e171abb0cb5e6daff3b65c2986efefa10f915e06f49dd55e04dd36854R57-R62)). we could potentially add a `replace` option, or just allow this to slide with an `eslint-disable` like you have.

Reply:
I'll write a `replace` function in EventMetaLib. The other thing I've been thinking about is getting the type of the EventMeta off of the event, which could have a function for it here as well.